content/backend/graphql-node-typescript/3-graphql-server.md

```diff
@@ -0,0 +1,242 @@
+
+  server.route({
+    method: "POST",
+    url: "/graphql",
+    handler: async (req, reply) => {
+      const request: Request = {
+        headers: req.headers,
+        method: req.method,
+        query: req.query,
+        body: req.body,
+      };
+
+      const { operationName, query, variables } = getGraphQLParameters(request);
+
+      const result = await processRequest({
+        request,
+        schema,
+        operationName,
+        query,
+        variables,
+      });
+
+      if (result.type === "RESPONSE") {
+        reply.send(result.payload);
+      } else {
+        reply.send({ error: "Stream not supported at the moment" });
+      }
+    }
+  });
+
+  server.listen(3000, "0.0.0.0", () => {
+    console.log(`Server is running on http://localhost:3000/`);
+  });
+}
+
+main();
+```
+
+</Instruction>
+
+You can now try to run your server, and you should be able to run our query in a HTTP-based GraphQL request. 
+
+<Instruction>
+
+If you have `curl` installed on your machine, you should be able to run and see the GraphQL response (you can also use any kind of HTTP requests tools, like [Postman](https://www.postman.com/)):
+
+```bash
+curl -X POST http://localhost:3000/graphql -H "Content-type: application/json" --data-raw '{"query": "query { info }"}'
+```
+</Instruction>
+
+So let's break down the code snippet for the server:
+
+1. Your created an new HTTP request handler, of type `POST`, based on the `/graphql` path. 
+1. The handler takes the incoming raw HTTP request and builds a `Request` object out of it.
+1. Then, the `getGraphQLParameters` method of `graphql-helix` extracts all the neccesary information from the raw request. 
+1. The `processRequest` gets all the input parameters, and runs the GraphQL request pipeline for us.
+1. The result of the GraphQL operation is being returned and sent to the client.  
+
+> GraphQL-Helix supports real-time GraphQL and stream-based responses, that's why your code need to check it's result before sending it. At the moment, our server only supports regular responses (`type="RESPONSE"`). You'll later add support for more transports.  
+
+### The complete GraphQL execution pipeline
+
+A complete GraphQL flow is consists of the following steps:
+
+1. Schema building - first, make sure that you have a GraphQL schema ready-to-use (built as executable schema). 
+1. `parse` - takes the GraphQL operation and parse it into a `DocuemntNode`.
+1. `validate` - uses the parsed GraphQL operation with the GraphQL schema, and validate it to make sure it matches the schema.
+1. Variables - GraphQL operations allow you to pass variables, which are just JSON object. 
+1. Context building - GraphQL allow you to pass a custom object, called `context`, and this object will be available for your resolvers. 
+1. `execute` - takes the operation and runs the resovlers based on the selection-set.
+
+With `graphql-helix`, this flow becomes simpler because you can just use `processRequest`.
```

Comment:
Why does it become easier? What `processRequest` actually does is not clarified.